enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,358 @@
+- Creating Managed Identity infrastructure with federated credentials for OpenShift operator `ServiceAccounts` (identified by namespace & name) and to output secrets containing the `clientID` of the Managed Identity to be provided as manifests for the installer. This command will process `CredentialsRequest` custom resources to identify service accounts that will be associated with Managed
+  Identities in Azure as federated credentials. For self-managed installation, `CredentialsRequests` will be extracted from the release image.
+
+```sh
+$ ./ccoctl azure
+Creating/updating/deleting cloud credentials objects for Azure
+
+Usage:
+  ccoctl azure [command]
+
+Available Commands:
+  create-all                Create key pair, identity provider and Azure Managed Identities
+  create-identity-provider  Create identity provider
+  create-key-pair           Create a key pair
+  create-managed-identities Create Azure Managed Identities
+  delete                    Delete Azure identity provider and Managed Identity infrastructure
+
+Flags:
+  -h, --help   help for azure
+
+Use "ccoctl azure [command] --help" for more information about a command.
+```
+
+#### Azure CredentialsRequests ServiceAccounts
+
+`CredentialsRequests` for the Azure platform must now list
+[ServiceAccountNames](https://github.com/openshift/cloud-credential-operator/blob/1f7a2602bf8a9ddec5d8fc29f77215697d9e7c07/pkg/apis/cloudcredential/v1/types_credentialsrequest.go#L57-L62)
+in order to for `ccoctl` to be able to create federated credentials
+for an Azure Managed Identity that are associated with the `name` and
+`namespace` of the `ServiceAccount`. Example:
+[aws-ebs-csi-driver-operator](https://github.com/openshift/cluster-storage-operator/blob/f1ddb697afb3c33d6d45936e58fad101abe26f13/manifests/03_credentials_request_aws.yaml#L11-L13).
+
+#### Credentials secret
+
+OpenShift operators currently obtain their long-lived credentials from a config secret with the following format:
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_client_secret: <client secret>
+  azure_region: <region>
+  azure_resource_prefix: <resource group prefix eg. "abutcher-az-t68n4">
+  azure_resourcegroup: <resource group eg. "abutcher-az-t68n4-rg">
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+We propose that when utilizing Azure Workload Identity, the credentials secret will contain an `azure_client_id` that is the `clientID` of the Managed Identity provisioned by `ccoctl` for the operator. The `azure_client_secret` key will be absent and instead we can provide the path to the mounted `ServiceAccount` token as an `azure_federated_token_file` key; the path to the mounted token is well
```

Comment:
To be sure I'm understanding, this means that the secret doesn't contain any secret information.  This is ok with me, but I'd like to state that as an explicit goal.  If this KEP is successful, the secrets in the cluster should not contain any secret information and all secrets should be presented as bound tokens, correct?

Reply:
Correct. No secret information will be present in the secrets.